frontstage/controllers/validators.py

```diff
@@ -0,0 +1,56 @@
+import itertools
```

Comment:
This whole thing isn't used.  It can be deleted (and thank god, it's an over engineered way to do validation)